api/search.rb

```diff
@@ -1,58 +1,76 @@
 def get_thread_ids(context, group_ids, local_params, search_text)
-  filters = []
-  filters.push({term: {commentable_id: local_params['commentable_id']}}) if local_params['commentable_id']
-  filters.push({terms: {commentable_id: local_params['commentable_ids'].split(',')}}) if local_params['commentable_ids']
-  filters.push({term: {course_id: local_params['course_id']}}) if local_params['course_id']
+  must = []
+  filter = []
+  must.push({term: {commentable_id: local_params['commentable_id']}}) if local_params['commentable_id']
+  must.push({terms: {commentable_id: local_params['commentable_ids'].split(',')}}) if local_params['commentable_ids']
+  must.push({term: {course_id: local_params['course_id']}}) if local_params['course_id']
+  must.push(
+    {
+      multi_match: {
+        query: search_text,
+        fields: [:title, :body],
+        operator: :AND
+      }
+    }
+  )
+  group_id = local_params['group_id']
+
+  if group_id
+    filter.push(
+      {
+        bool: {
+          should: [
+            {:bool => {:must_not => {:exists => {:field => :group_id}}}},
+            {:term => {:group_id => group_id}}
+          ]
+        }
```

Comment:
Do we need a minimum should match option here?

Reply:
Thanks, I try to simplify this query and it works

Reply:
Thanks, I try to simplify this query and it works